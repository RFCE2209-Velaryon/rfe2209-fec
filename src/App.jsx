import React from 'react';
import $ from 'jquery';

import Product from './components/productDetail/productDetail.jsx';
import QuestionsAndAnswers from './components/Q&A/qna.jsx';
import Related from './components/relateANDcompare/relateANDcompare.jsx';
import Reviews from './components/reviews/reviews.jsx';
import Products from './components/productDetail/productDetail.jsx';

function App () {
  const [product, setProduct] = React.useState({name:'product1'});

  // useEffect on a blank array is used to run the inside code once upon page load
  React.useEffect(() => {
    // TODO: make an API call to get one product
  }, []);

  return (
    <div>
      <Product />
      <Related />
      <QuestionsAndAnswers />
      <Reviews />
    </div>
  )
}

export default App;