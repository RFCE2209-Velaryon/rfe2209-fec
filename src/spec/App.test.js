import renderer from 'react-test-renderer';
import React from 'react';

import App from '../App.jsx';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
 });