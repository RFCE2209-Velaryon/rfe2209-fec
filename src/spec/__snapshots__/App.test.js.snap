// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Product Detail Component
  </h1>
  <h1>
    Related Items Component
  </h1>
  <h1>
    Q&A Component
  </h1>
  <h1>
    Review Component
  </h1>
</div>
`;
